cloudsync/sync/manager.py

```diff
@@ -1388,6 +1398,29 @@ def embrace_change(self, sync, changed, synced):  # pylint: disable=too-many-ret
         log.debug("nothing changed %s", sync)
         return FINISHED
 
+    def handle_changed_is_missing(self, sync, changed, synced):
+        if (sync[synced].exists == EXISTS and
+            #not sync[synced].changed and #TODO: was in Mike's original change
+            self.providers[changed].paths_match(sync[changed].path, sync[changed].sync_path, for_display=True) and
```

Comment:
Can you add a comment explaining why we need to check if the path and hash match the sync_path and sync_hash? I'm having trouble following the logic

Reply:
ended up removing those checks, it doesn't look like they are necessary